frontend/dockerfile/dockerfile_heredoc_test.go

```diff
@@ -112,6 +113,72 @@ COPY --from=build /dest /
 	}
 }
 
+func testCopyHeredocSpecialSymbols(t *testing.T, sb integration.Sandbox) {
+	f := getFrontend(t, sb)
+
+	dockerfile := []byte(`
+FROM scratch
+
+COPY <<EOF quotefile
+"quotes in file"
+EOF
+
+COPY <<EOF slashfile1
+\
```

Comment:
Could we have a case with `<<'EOF'` as well to verify that `\` is retained in that case. If these cases already exist in other tests then comment pointing to these cases from here would do.